### PR TITLE
CLN: fix compiler warnings in tzconversion.pyx

### PR DESCRIPTION
**Pull request by ghost**
Fixes 
```
pandas/_libs/tslibs/tzconversion.c:3157:21: warning: ‘__pyx_pybuffernd_dst_hours.diminfo[0].strides’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 3157 |   __Pyx_LocalBuf_ND __pyx_pybuffernd_dst_hours;
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~
pandas/_libs/tslibs/tzconversion.c:3149:21: warning: ‘__pyx_pybuffernd_ambiguous_array.diminfo[0].strides’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 3149 |   __Pyx_LocalBuf_ND __pyx_pybuffernd_ambiguous_array;

```

### CODE REVIEW

File: pandas/_libs/tslibs/tzconversion.pyx [hunk 1]
```diff
@@ -167,7 +169,8 @@ timedelta-like}
         # where result_a != result_b and neither of them are NAT)
         both_nat = np.logical_and(result_a != NPY_NAT, result_b != NPY_NAT)
         both_eq = result_a == result_b
-        trans_idx = np.squeeze(np.nonzero(np.logical_and(both_nat, ~both_eq)))
+        trans_idx = np.squeeze(np.nonzero(np.logical_and(both_nat,
+                                                        np.logical_not(both_eq))))
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/tzconversion.pyx]: I don’t think the numpy functions work on memoryviews

Comment (pilkibun) [reply pandas/_libs/tslibs/tzconversion.pyx]: 2nd time that tests pass locally and fail on CI. I'm working on other cython changes and can see the effects after recompilation. It's a mystery to me. push another commit.

**Comment by jbrockmendel** [comment]
LGTM

**Comment by jreback** [comment]
thanks

File: pandas/_libs/tslibs/tzconversion.pyx (hunk 1 above)
Comment (pilkibun) [reply pandas/_libs/tslibs/tzconversion.pyx]: > I don’t think the numpy functions work on memoryviews

I found the problem, `npy_bool` memoryviews are not supported in Cython 0.29.
This is fixed in the cython's master branch, which I had installed.

**Pull request merged**